README.md

```diff
@@ -57,33 +53,43 @@ The `esb.yml` file is the main configuration of your ESB application, where you
 ```yaml
 services:
   _defaults:
-    autowire: true                  # This is optional (see https://symfony.com/doc/current/service_container/autowiring.html)
+    autowire: true                    # This is optional (see https://symfony.com/doc/current/service_container/autowiring.html)
 
-  My\Esb\Producer:                  # A producer service definition
+  My\Esb\Producer:                    # A producer service definition
     arguments: []
 
-  My\Esb\Worker:                    # A worker service definition
+  My\Esb\Worker:                      # A worker service definition
     arguments: []
 
 
 flows:
-  sample_flow:                      # The flow "code" and will be the Beanstalkd tube name
-    description: Sample Flow        # The flow description
+  sample_flow:                        # The flow "code" and will be the Beanstalkd tube name
+    description: Sample Flow          # The flow description
     producer:
-      service: My\Esb\Producer      # A producer service ID defined above
+      service: My\Esb\Producer        # A producer service ID defined above
     worker:
-      service: My\Esb\Worker        # A worker service ID defined above
-      instances: 1                  # The number of worker instances to spawn for this flow
-      release_delay: 0              # The jobs release delay in seconds for this flow (see the Beanstalkd protocol here https://github.com/beanstalkd/beanstalkd/blob/master/doc/protocol.txt)
-      max_retry: 5                  # The number of maximum work retries for a job in this tube/flow before being buried
-    depends_on: ['other_flow']      # Optional: dependencies of this flow toward other flows
-
+      service: My\Esb\Worker          # A worker service ID defined above
+      instances: 1                    # The number of worker instances to spawn for this flow
+      release_delay: 0                # The jobs release delay in seconds for this flow (see the Beanstalkd protocol here https://github.com/beanstalkd/beanstalkd/blob/master/doc/protocol.txt)
+      max_retry: 5                    # The number of maximum work retries for a job in this tube/flow before being buried
+    depends_on: ['other_flow_1', 'other_flow_2']  # Optional: dependencies of this flow toward other flow(s)
+    delay_after_idle_time: 1000       # Delay that a worker with dependencies waits before working the first job received after the tube was empty
+    initial_polling_interval: 1000    # Initial polling delay that a worker waits when it has to wait for a dependency that is not idle
+    maximum_polling_interval: 60000   # Maximum polling delay that a worker waits when it has to wait for a dependency that is not idle
+    polling_interval_multiplier: 2    # Polling delay increase factor whenever a worker is waiting for a dependency that is not idle
+
```

Comment:
@azambon maybe it's better to group all "dependencies" stuff under the same key. What do you think?
Something like:
```yaml
    dependencies:
        flows: ['other_flow_1', 'other_flow_2']
        delay_after_idle_time: 1000
        initial_polling_interval: 1000
        maximum_polling_interval: 60000
        polling_interval_multiplier: 2
```

I think that this way we can require those parameters only if the `dependencies` key is set and remove the default values.

Reply:
Honestly I think that the defaults are reasonable for a wide range of applications. One shouldn't need to change them onless he/she is doing something very specific.
Forcing to specify all those timing parameters just because you need to declare a dependency seems a bit verbose, error-inducing and, frankly, useless.

Reply:
Ok we can keep the default values but having them under the same `dependencies` key enforce the concept that they are related to the flow dependencies feature. @fabianaromagnoli what do you think?